training/src/main/scala/com/airbnb/aerosolve/training/photon/ml/data/ConvertExamplesToPhotonMLAvroJob.scala

```diff
@@ -0,0 +1,176 @@
+    } else {
+      ConfigFactory.load(configPath.getPath(),
+        ConfigParseOptions.defaults.setAllowMissing(false),
+        ConfigResolveOptions.defaults)
+    }
+
+    val sparkConf = new SparkConf().setAppName("Convert Thrift Example to PhotonML Avro")
+    val sc = new SparkContext(sparkConf)
+
+    convertExamplesToPhotonGAMEAvro(sc, config)
+  }
+
+  private def getLabelFromExample(example: Example): Double = {
+    if (example.getExample().get(0).floatFeatures.get(GenericPipeline.LABEL).get("") > 0) {
+      1d
+    } else {
+      0d
+    }
+  }
+
+  /**
+    * This method converts an RDD of examples to PhotonML GAME Avro formats and save to HDFS.
+    * Could be used as an entry-point in programming.
+    *
+    * @param sc         The spark context
+    * @param taskConfig config of the task
+    */
+  def convertExamplesToPhotonGAMEAvro(sc: SparkContext,
+                                      taskConfig: Config): Unit = {
+    val (transformedData, avroSchema) = loadExamples(sc, taskConfig)
+
+    val outputPath = new Path(taskConfig.getString("output_path"))
+
+    val fs = outputPath.getFileSystem(new Configuration())
+    fs.delete(outputPath, true)
+
+    val avroSchemaStr = avroSchema.toString()
+    logger.info(s" Configured output Avro schema: ${avroSchema.toString(true)}")
+
+    val metaDataExtractorClass = Try(taskConfig.getString("meta_data_extractor_class"))
+      .getOrElse(classOf[SingleFloatFamilyMetaExtractor].getName())
+
+    val metaDataExtractorContext = Try(taskConfig
+      .getConfig("meta_data_extractor_context").entrySet().asScala
+      .map { case entry => (entry.getKey(), entry.getValue().toString()) }.toMap
+    ).getOrElse(immutable.Map[String, String]())
+
+    val trainingExampleAvros = transformedData.mapPartitions { case iter =>
+      val schema = new Schema.Parser().parse(avroSchemaStr)
+
+      val metaDataExtractor = Class.forName(metaDataExtractorClass)
+        .newInstance().asInstanceOf[ExampleMetaDataExtractor]
+
+      iter.map { case example =>
+        val record = new GenericData.Record(schema)
+
+        val label = getLabelFromExample(example)
+
+        // TODO decide which key to provide for training example, currently this is not a must-have field
+        // record.put(AVRO_UID, key)
+        record.put(AVRO_LABEL, label)
+        record.put(AVRO_META_DATA_MAP,
+          metaDataExtractor.buildMetaDataMap(example, metaDataExtractorContext))
+
+        createAvroFeatureArrays(record, example, schema)
+
+        record
+      }
+    }
+
+    val job = Job.getInstance
+    AvroJob.setOutputKeySchema(job, avroSchema)
+    val hadoopConf = job.getConfiguration()
+
+    // Compress Avro binary files
+    hadoopConf.set("mapreduce.output.fileoutputformat.compress", "true")
+    hadoopConf.set(AvroJob.CONF_OUTPUT_CODEC, "deflate")
+
+    // Save as Avro records
+    trainingExampleAvros
+      .map { r => (new AvroKey(r), NullWritable.get()) }
```

Comment:
so GenericData.Record has to be wrap by AvroKey?  what's the use for NullWritable.get() here?